broker/src/main/java/io/camunda/zeebe/broker/clustering/topology/ClusterTopology.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.clustering.topology;
+
+import com.google.common.collect.ImmutableMap;
+import io.atomix.cluster.MemberId;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Represents the cluster topology which describes the current active, joining or leaving brokers
+ * and the partitions that each broker replicates.
+ */
+public record ClusterTopology(
+    long version, Map<MemberId, MemberState> members, ClusterChangePlan changes) {
+
+  static ClusterTopology init() {
+    return new ClusterTopology(0, Map.of(), ClusterChangePlan.empty());
+  }
+
+  ClusterTopology addMember(final MemberId memberId, final MemberState state) {
+    if (members.containsKey(memberId)) {
+      throw new IllegalStateException(
+          String.format(
+              "Expected add a new member, but member %s already exists with state %s",
+              memberId.id(), members.get(memberId)));
+    }
+
+    final var newMembers =
+        ImmutableMap.<MemberId, MemberState>builder().putAll(members).put(memberId, state).build();
+    return new ClusterTopology(version, newMembers, changes);
```

Comment:
It is not necessary. We will increment the version when we initiate configuration changes. While developing the concept, I found no value increasing the version here for each update, since the version in MemberState is already enough to resolve conflicts. 
Is the confusing? May be it should be documented better.

Reply:
Ah okay, that makes sense :+1: I'd say we should document it, yeah.